x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java

```diff
@@ -82,7 +84,7 @@
     }
 
     long ramBytesUsed() {
-        return trainedModelDefinition.ramBytesUsed();
+        return modelSize;
```

Comment:
```suggestion
        // This should always be cached and not calculated on call. 
        // This is because the caching system calls this method on every promotion call that changes the LRU head
        // Consequently, recalculating can cause serious throughput issues due to LRU changes in the cache
        return cachedRamBytesUsed;
```